### PR TITLE
[SPARK-31655][BUILD][3.0] Upgrade snappy-java to 1.1.7.5

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

snappy-java have release v1.1.7.5, upgrade to latest version.

Fixed in v1.1.7.4
- Caching internal buffers for SnappyFramed streams #234
- Fixed the native lib for ppc64le to work with glibc 2.17 (Previously it depended on 2.22)

Fixed in v1.1.7.5
- Fixes java.lang.NoClassDefFoundError: org/xerial/snappy/pool/DefaultPoolFactory in 1.1.7.4

https://github.com/xerial/snappy-java/compare/1.1.7.3...1.1.7.5

v 1.1.7.5 release note:
https://github.com/xerial/snappy-java/commit/edc4ec28bdb15a32b6c41ca9e8b195e635bec3a3
### Why are the changes needed?
Fix bug 

### Does this PR introduce _any_ user-facing change?
No 

### How was this patch tested?
No need

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Do you think we can have a valid unit test case for this? (non-ppc64le part)

**Comment by dongjoon-hyun** [comment]
BTW, currently, Jenkins jobs seems to be broken due to some outside issues.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/

**Comment by AngersZhuuuu** [comment]
> BTW, currently, Jenkins jobs seems to be broken due to some outside issues.
> 
> * https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/

No wonder this strange mistake

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by srowen** [comment]
This needs to go into master first, right?

**Comment by AngersZhuuuu** [comment]
> This needs to go into master first, right?

Has been merged into master
https://github.com/apache/spark/pull/28472

Backport to 2.4 & 3.0 now.

Seems jenkins have some problem

**Comment by srowen** [comment]
Does test-dependencies.sh pass locally? I don't see any clear output here.

**Comment by AngersZhuuuu** [comment]
> Does test-dependencies.sh pass locally? I don't see any clear output here.

Yeah, in local  can pass it.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by AngersZhuuuu** [comment]
> Retest this please.

Do you know what happened? seems in master mvn is different from branch-3.0 and branch-2.4

In master's succes build 
```
exec: curl --silent --show-error -L https://downloads.lightbend.com/zinc/0.3.15/zinc-0.3.15.tgz
exec: curl --silent --show-error -L https://downloads.lightbend.com/scala/2.12.10/scala-2.12.10.tgz
exec: curl --silent --show-error -L https://www.apache.org/dyn/closer.lua?action=download&filename=/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz
Using `mvn` from path: /home/jenkins/workspace/Spark............
..............
```

**Comment by AngersZhuuuu** [comment]
In auto check, `test-deps.sh` success.

**Comment by dongjoon-hyun** [comment]
There is a network outage in the Jenkins farm.
Please see the current status.

- https://issues.apache.org/jira/browse/SPARK-31693

For the above maven difference, it should be the same because I fixed it in SPARK-31691 .

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
FYI, the dependency failure will be fixed by the following.
- https://github.com/apache/spark/pull/28532

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by maropu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
@dongjoon-hyun Seems something still wrong, missing something on branch-3.0?

**Comment by dongjoon-hyun** [comment]
Yes. `amp-jenkins-worker-05` server is still broken. If you click the Jenkins log, you can see that it failed on ` amp-jenkins-worker-05` in 10 minutes.

**Comment by dongjoon-hyun** [comment]
BTW, the previous failed UT, `support multiple barrier() call within a single task`, is disabled by @cloud-fan now. I merged it. Let's test again.

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by srowen** [comment]
Hm, I wonder if `mvn -U dependency:purge-local-repository` would help, like opening a PR that just tries changing this in the test script, to see if it can be run once on the offending machine (and then not merge it)

**Comment by dongjoon-hyun** [comment]
Thank you for the tip. Yes. It's possible, but in a shared cluster, is it okay to purge it by one Maven job while the other Maven jobs are working? Maybe, we can enforce it at the specific single machine by recognizing the machine name because the machine seems to fail already always in any way.

**Comment by srowen** [comment]
Hm, not sure. I think it's thread-safe. At least, if it doesn't work at all now, can't be much worse. This would not be merged, just a one-time thing.

**Comment by dongjoon-hyun** [comment]
Got it. I'll try to make a PR to purge it.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to branch-3.0.
cc @rxin and @yhuai since this is a dependency change in 3.0.

**Pull request closed without merging**
